### ISSUE TITLE
Documentation of `interface` (Containers/Variables/...)

### ISSUE THREAD

**Issue by pgrete** [lanl/parthenon#66]
During the discussion of #64 I noticed that I'm less aware of what's going on in many of the features in the `interface` folder than I'd like to.
Is there any internal documentation that you can share?
I tried to go through the code, but quite a few piece of commented out so I'm not sure if those are not there because they are unused/unfinished/defunct or sth else.

One particular piece I'm interested in (which is also relevant to additional upcoming changes in Kokkosizing AMR functionality), is what the `flux` and `coarse` arrays are used in a `Variable` and how far this has implication to AMR/communication.

As far as I was able to follow the discussion at the workshop, the `Container` would be used to hold `Variable`s plus some metadata and `Variables` hold "raw" data (plus some metadata).
This would then allow for more flexibility, e.g., in the output routines ("dump a slice of the first derivative of all  'passive scalar' type variables").
The current feature set (or what I deduced from the code) seems to go way beyond that, e.g., also with respect to handling stages.

Bottom line: I'd be great if we could discuss/document/... pieces in the `interface` folder to know
- in how far this affects the Kokkosization of the remaining piece (particularly AMR)
- what features are available and how to use them from a user (here application developer) point of view.

**Comment by jdolence** [comment]
closed via #75 and #76
